File: youlexuan_cart_interface/src/main/java/com/offcn/cart/service/CartService.java
package com.offcn.cart.service;

import com.offcn.pojo.group.CartGroup;

import java.util.List;

public interface CartService {


    List<CartGroup> addItemToCartList(List<CartGroup> cartList, Long itemId, Integer num);

    List<CartGroup> findCartListFromRedis(String username);

    List<CartGroup> mergeRedisAndCookie(List<CartGroup> redisCartList, List<CartGroup> cartList);

    void saveListToRedis(String username, List<CartGroup> redisCartList);
}

File: youlexuan_cart_web/src/main/webapp/js/service/cartService.js
//用户表服务层
app.service('cartService',function ($http) {
    this.findCartList = function () {
        return $http.get('../cart/findCartList.do');
    }
    this.addItemToCartList = function (itemId,num) {
        return $http.get('../cart/addItemToCartList.do?itemId='+itemId+"&num="+num);
    }
    this.findAddressByLoginUser = function () {
        return $http.get('../address/findAddressByLoginUser.do')
    }
    this.submitOrder=function (order) {
        return $http.post('../order/submitOrder.do',order);
    }
})